wandb/sdk/launch/runner/local_container.py

```diff
@@ -139,9 +140,13 @@ def run(
 
         if launch_project.docker_image:
             if image_uri.endswith(":latest") or not docker_image_exists(image_uri):
-                pull_docker_image(image_uri)
+                try:
+                    pull_docker_image(image_uri)
+                except Exception as e:
```

Comment:
A few thoughts:
1. Shouldn't the `Registry` be involved here? (Probably a future better fix)
2. Let's try a pull, warn then if the image doesn't exist throw an error

Reply:
So for (2) are you just proposing that I add a warn before we try to pull the image?

For (1) I see your point but I don't want to force a user to configure a registry for any remote image that they want to run. For instance maybe I want to `wandb launch -d ...` an image in a private registry for which I have already configured docker credentials in my os keychain. The reason we have these registry configs is really for distributed environments like k8s where these credentials need to be passed around by the agent. For local container I think it's reasonable to make the user responsible for registry credentials.

Reply:
For 2 I just propose warning, and only erroring if the image does not exist after the pull is attempted. If the image doesn't exist at that point its worth erroring